### ISSUE TITLE
infix/sufix search

### ISSUE THREAD

**Issue by sdargutev** [dakaraphi/eclipse-plugin-commander#35]
Adding shortcuts for following search strategies would be nice:

**infix search** - prioritize matches that contain the search term inside the name. See screenshot, The _ALocal.java_ should be on top with an infix search.

![unbenannt](https://user-images.githubusercontent.com/21322759/33214523-99f11136-d12c-11e7-8436-1f8493e88084.PNG)

**suffix search** - prioritize matches that contain the given term at the end

One more nice to have would be to lower the rank of **.class** files. It is unlikely that someone searches for a **.class** file. And there are lots and lots of them...

**Comment by dakaraphi** [comment]
Actually inclusion of `class` files was one of the things I've been asked for the most.
It was just added in last update.  It basically merges eclipse's `open resource` and `open type` into one.

You can do this `searchterm!class` or `searchterm,!class` to be even more specific to eliminate classes from the results.

I could consider maybe either some type of quick toggle to include or exclude classes, or a way to set a preference.

**Comment by sdargutev** [comment]
oh, i guess different people different needs :) 
A simple Toggle would be nice. 

And thank you for pointing out how i can exclude them!

**Comment by paulvi** [comment]
@dakaraphi Should it be using some sort of * if you know you are looking for a part of word, e.g.`*Local`

**Comment by dakaraphi** [comment]
@paulvi 
> Should it be using some sort of * if you know you are looking for a part of word, e.g.*Local

1.  That would be inconsistent with the behavior of command palettes in general
1.  Searching for part of a word is a common primary use case
1.  Specifying `*` would be an extra character and action the user would need to take

**Comment by paulvi** [comment]
Then for a case brought above `ALocal` would not and should not appear above all `Local*`,
just because we write and type from left to right.

Generally infix/suffix search, as this issue is titled, require some way to specify search within middle of words or at ends. If not `*` than should be some other key.

**Comment by dakaraphi** [comment]
> Then for a case brought above ALocal would not and should not appear above all Local*

I agree.  I consider this a feature request for specialized searches.  So I would not alter the default behavior.

There is a limit to the search features a user can specify that can be both fast and easy for the user to enter.  I've considered allowing some way for a user to optionally specify a regex for matching so they can do edge case matches that should cover pretty much any specialized cases.
